lib/fastlane/plugin/wpmreleasetoolkit/actions/ios/ios_check_beta_deps.rb

```diff
@@ -1,25 +1,50 @@
+require 'yaml'
+
 module Fastlane
   module Actions
     class IosCheckBetaDepsAction < Action
+      ALL_PODS_STABLE_MESSAGE = 'All pods are pointing to a stable version. You can continue with the code freeze.'.freeze
+      NON_STABLE_PODS_MESSAGE = "Please create a new stable version of those pods and update the Podfile to the newly released version before continuing with the code freeze:\n".freeze
+
       def self.run(params)
         require_relative '../../helper/ios/ios_version_helper'
         require_relative '../../helper/ios/ios_git_helper'
 
-        beta_pods = []
-        File.open(params[:podfile]).each do |li|
-          beta_pods << li if li.match('^\s*\t*pod.*beta')
+        yaml = YAML.load_file(params[:lockfile])
+        non_stable_pods = {} # Key will be pod name, value will be reason for flagging
+
+        # Find pods referenced by commit and branch to a repo. if any
+        yaml['EXTERNAL SOURCES']&.each do |pod, options|
+          non_stable_pods[pod] = 'commit' if params[:report_commits] && options.key?(:commit)
+          non_stable_pods[pod] = 'branch' if params[:report_branches] && options.key?(:branch)
+        end
+
+        # Find pods whose resolved version matches the regex
+        version_pattern = params[:report_version_pattern]
+        unless version_pattern.nil? || version_pattern.empty?
+          regex = begin
+            Regexp.new(version_pattern)
+          rescue RegexpError
+            UI.user_error!("Invalid regex pattern: `#{version_pattern}`")
```

Comment:
Side note. I wanted to share appreciation for this kind of error handling where the things that's wrong is clearly reported in the error message. Recently, I've bump my head against some vague error messages; seeing this is encouraging. Thanks!